postgres-14/Dockerfile

```diff
@@ -13,16 +13,21 @@ LABEL maintainer="Citus Data https://citusdata.com" \
 
 ENV CITUS_VERSION ${VERSION}.citus-1
 
-# install Citus
-RUN apt-get update \
-    && apt-get install -y --no-install-recommends \
-       ca-certificates \
-       curl \
+#install postgresql repo and citus
```

Comment:
I do not like the fact that we have different steps in these Dockerfiles. This should not be necessary for several reasons.

1. The base image already has the PGDG repo installed. No reason to have explicit steps here.
2. If we keep introducing new Dockerfiles, we are increasing the technical debt and making it harder to maintain this repo.


Please consider the following `RUN` command if it looks ok with you:

```Dockerfile
# install Citus
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
       ca-certificates \
       curl \
    && curl -s https://install.citusdata.com/community/deb.sh | bash \
    && apt-get install -y postgresql-$PG_MAJOR-citus-10.2=$CITUS_VERSION \
                          postgresql-$PG_MAJOR-hll=2.16.citus-1 \
                          postgresql-$PG_MAJOR-topn=2.4.0 \
    && apt-get purge -y --auto-remove curl \
    && rm -rf /var/lib/apt/lists/*
```

I just copied the same steps from another Dockerfile from this repo. LMK if that's ok with U

Reply:
Other than pg14 it is OK. Postgres repo installation differs from others

Reply:
For latest and postgres 12 the block is same as you suggested so no need to change I think

Reply:
I know that the repo installation is different on pg14. However the base image already handles that, and we do not need to do provide any extra steps as long as our base image is `postgres:14beta3`

Reply:
Thank you for the heads up. I didn't think that base image could have such definitions now fixed as you suggested